### PR TITLE
feat(clients/java): add user-agent to java client requests

### LINKED ISSUE #4265: Java client should send the client type and version with his auth request
**Is your feature request related to a problem? Please describe.**

On camunda cloud we have sometimes bug reports / incidents which might be already fixed with the latest version of the client, but the user uses a old client version.

**Describe the solution you'd like**

We would like to detect the client type and version, on requesting the auth token. This would make incident investigation much easier, since we then don't need to ask the user for this kind of information if there went something wrong. It would also make it possible to correlate failures with know bugs.

We thought about adding this kind of information in the user agent header on the client HTTP request. 

**Describe alternatives you've considered**
The alternative would be to ask the user for this kind of information.

**Additional context**

The content of the user agent should at least contain:

 * client type, e.g. java client
 * client version, e.g. 0.22.2

### PR DESCRIPTION
**Pull request by miguelpires**
## Description

Add user-agent header with client and version information to gRPC and OAuth requests.

## Related issues

closes #4265 

## Pull Request Checklist

- [ ] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [ ] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [ ] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientImpl.java [hunk 1]
```diff
@@ -123,6 +124,7 @@ public static ManagedChannel buildChannel(final ZeebeClientConfiguration config)
 
     configureConnectionSecurity(config, channelBuilder);
     channelBuilder.keepAliveTime(config.getKeepAlive().toMillis(), TimeUnit.MILLISECONDS);
+    channelBuilder.userAgent("client: java, version: " + VersionUtil.getVersion());
```
Comment (saig0) [inline clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientImpl.java:127]: Same as for the `OAuthCredentialsProvider`.

File: clients/java/src/main/java/io/zeebe/client/impl/Loggers.java [hunk 2]
```diff
@@ -24,4 +24,5 @@
       LoggerFactory.getLogger("io.zeebe.client.job.poller");
   public static final Logger JOB_WORKER_LOGGER =
       LoggerFactory.getLogger("io.zeebe.client.job.worker");
+  public static final Logger VERSION_LOGGER = LoggerFactory.getLogger("io.zeebe.client.util");
```
Comment (saig0) [inline clients/java/src/main/java/io/zeebe/client/impl/Loggers.java:27]: The logger package looks a bit uncommon with `.util`. Maybe, we can have a logger for general or managing things `io.zeebe.client`.

**Review by saig0 — COMMENTED** [review]
Looks good. Just a few comments.

File: clients/java/src/main/java/io/zeebe/client/impl/oauth/OAuthCredentialsProvider.java [hunk 3]
```diff
@@ -162,6 +163,8 @@ private ZeebeClientCredentials fetchCredentials() throws IOException {
     connection.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
     connection.setRequestProperty("Accept", "application/json");
     connection.setDoOutput(true);
+    connection.setRequestProperty(
+        "User-Agent", "client: java, version: " + VersionUtil.getVersion());
```
[the remark this answers is not in the document]
Comment (MiguelPires) [reply clients/java/src/main/java/io/zeebe/client/impl/oauth/OAuthCredentialsProvider.java:167]: I'd happily change it but this is the format used in the C# and node clients so I figured it was better to stick to one format. Do you think it's better to break from that and change it here and in the Go clients?

[the remark this answers is not in the document]
Comment (saig0) [reply clients/java/src/main/java/io/zeebe/client/impl/oauth/OAuthCredentialsProvider.java:167]: You're right. I also prefer one format for all clients. But I would also prefer a more common format. 

Let's shortly ask the maintainers @Zelldon and @jwulf what they are thinking of changing the user-agent` format?

[the remark this answers is not in the document]
Comment (saig0) [reply clients/java/src/main/java/io/zeebe/client/impl/oauth/OAuthCredentialsProvider.java:167]: @jwulf @Zelldon :bellhop_bell: Your input is requested.

[the remark this answers is not in the document]
Comment (Zelldon) [reply clients/java/src/main/java/io/zeebe/client/impl/oauth/OAuthCredentialsProvider.java:167]: I fine with that. In the end it doesn't matter. It is just to find out which client is used at the cloud.

[the remark this answers is not in the document]
Comment (Zelldon) [reply clients/java/src/main/java/io/zeebe/client/impl/oauth/OAuthCredentialsProvider.java:167]: Please create then issue in the corresponding clients.

[the remark this answers is not in the document]
Comment (MiguelPires) [reply clients/java/src/main/java/io/zeebe/client/impl/oauth/OAuthCredentialsProvider.java:167]: Related issues: 
https://github.com/creditsenseau/zeebe-client-node-js/issues/170
https://github.com/zeebe-io/zeebe-client-csharp/issues/144

**Comment by MiguelPires** [comment]
Thanks for the feedback @saig0, I had to rebase on develop due to have the fix for expiring certs in the tests

**Review by saig0 — APPROVED** [review]
:+1:

**Pull request merged**